shared/src/main/scala/io/estatico/newtype/macros/NewTypeMacros.scala

```diff
@@ -75,22 +85,40 @@ private[macros] class NewTypeMacros(val c: blackbox.Context) {
     // Ensure we're not trying to inherit from anything.
     validateParents(clsDef.impl.parents)
     // Build the type and object definitions.
-    generateNewType(clsDef, modDef, valDef, tparamsNoVar, tparamNames, tparamsWild)
+    generateNewType(clsDef, modDef, valDef, tparamsNoVar, tparamNames, tparamsWild, subtype)
+  }
+
+  def mkBaseTypeDef(clsDef: ClassDef, reprType: Tree, subtype: Boolean) = {
+    val refinementName = TypeName(clsDef.name.decodedName.toString + "$newtype")
+    (clsDef.tparams, subtype) match {
+      case (_, false)      =>  q"type Base             = { type $refinementName } "
+      case (Nil, true)     =>  q"type Base             = $reprType"
+      case (tparams, true) =>  q"type Base[..$tparams] = $reprType"
+    }
   }
 
+  def mkTypeTypeDef(clsDef: ClassDef, tparamsNames: List[TypeName], subtype: Boolean) =
+    (clsDef.tparams, subtype) match {
+      case (Nil, false) =>     q"type Type             <: Base with Tag"
+      case (tparams, false) => q"type Type[..$tparams] <: Base with Tag[..$tparamsNames]"
+      case (Nil, true)  =>     q"type Type             =  Base with Tag"
+      case (tparams, true) =>  q"type Type[..$tparams] =  Base[..$tparamsNames] with Tag[..$tparamsNames]"
```

Comment:
Wouldn't this lead to problems with type inference again? What do we gain from using `=` instead of `<:`?

Reply:
I think this was the result of me thinking that the abstract type was causing boxing, but later I realized boxing was happening for a different reason (which you commented on as well -
 https://github.com/scala/bug/issues/10770).

Just added a commit to fix this and all looks good, nice catch!